src/accelerate/utils/deepspeed.py

```diff
@@ -19,6 +19,20 @@
 
 from ..optimizer import AcceleratedOptimizer
 from ..scheduler import AcceleratedScheduler
+from ..state import AcceleratorState
+from .dataclasses import DeepSpeedPlugin, DistributedType
+
+
+def get_active_deepspeed_plugin(state: AcceleratorState) -> DeepSpeedPlugin:
+    """
+    Returns the currently active DeepSpeedPlugin.
+
+    Raises:
+        ValueError: If DeepSpeed was not enabled and this function is called.
+    """
+    if state.distributed_type != DistributedType.DEEPSPEED:
+        raise ValueError("DeepSpeed was not enabled. This function should not be called.")
```

Comment:
could this be made more user-friendly, when one asserts it helps to guide the user to how the issue can be fixed. "This function should not be called" is not actionable. Possibly:

> Can't retrieve the active DeepSpeed plugin since none were enabled. Use <fill in> to enable one.

Reply:
Went with:

```python
        raise ValueError(
            "Couldn't retrieve the active `DeepSpeedPlugin` as none were enabled. "
            "Please make sure that either `Accelerator` is configured for `deepspeed` "
            "and make sure that a `DeepSpeedPlugin` has been enabled (`plugin.enable()`) before calling this."
        )
```

Which gets the point across

Reply:
super!